client/client.go

```diff
@@ -25,93 +26,86 @@ const (
 
 var (
 	logger               = log.New(os.Stdout, "", 0)
-	_             Client = (*GRPCClient)(nil)
+	_             Client = &GRPCClient{}
 	defaultClient Client
 	doOnce        sync.Once
```

Comment:
I think the desired behavior here with the doOnce/defaultClient is to provide a default client like a `http.DefaultClient`. Of course, the user is required to initialize and create it, so the value is a little less compared to `http.DefaultClient`. If we need to support this use case, then adding `WithDefault()` functional option that reuses the hidden defaultClient will bring back this behavior.